MbsAPI/f_stccomm.c

```diff
@@ -859,206 +843,193 @@ INTS4* pi_channel;
+        }
     }
 
     retval = -1;
 
-    bzero( (CHARS*) &s_server.sock_name , sizeof( s_server.sock_name ) );
-    s_server.sock_name.sin_family        = AF_INET;
-    s_server.sock_name.sin_addr.s_addr   = htonl(INADDR_ANY);
-    s_server.sock_name.sin_port          = htons(*pl_port);
-
-    retval = bind(  s_server.sock_rw,
-                    (struct sockaddr*) &s_server.sock_name,
-                    sizeof(s_server.sock_name));
-    if( retval == -1 ) {
-      close( s_server.sock_rw );
-
-      *pl_port += 1;
-
-      if( *pl_port > 65535 ) {
-        printf("--E--f_stc_createserver() portnumber exceeded > 655535\n");
-
-        switch( errno ) {
-        case EBADF       :
-          return STC__INVSOCK;
-        case ENOTSOCK    :
-          return STC__NOTSOCK;
-        case EADDRINUSE  :
-          return STC__PORTINUS;
-        case EINVAL      :
-          return STC__SOCKISC;
-        case EACCES      :
-          return STC__SOCKISP;
-        default          :
-          return STC__FAILURE;
-        }  /* end switch( errno )   */
-
-      }  /* end if *pl_port > ... )   */
-
-    } /* end if (retval == -1 )  */
-
-
-  }
-
-  retval = listen(s_server.sock_rw,5);
-  if( retval == -1 ) {
-    switch( errno ) {
-    case EBADF       :
-      return STC__INVSOCK;
-    case ENOTSOCK    :
-      return STC__NOTSOCK;
-    default          :
-      return STC__FAILURE;
+    while (retval == -1 && retry == 1) {
+
+        s_server.sock_rw = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
+
+        switch (s_server.sock_rw) {
+            case -1:
+                switch (errno) {
+                    case EAFNOSUPPORT:
+                        return STC__INVADDRF;
+                    case ESOCKTNOSUPPORT:
+                        return STC__SOCKNSUP;
+                    case EPROTONOSUPPORT:
+                        return STC__INVPROTO;
+                    case EMFILE:
+                        return STC__SOCKTABF;
+                    case ENOBUFS:
+                        return STC__SOCKSPAF;
+                    default:
+                        return STC__FAILURE;
+                }
+        }
+
+        retval = gethostname(s_server.hostname, sizeof(s_server.hostname));
+        if (retval) {
+            printf("--E--f_stc_createserver() error get local hostname\n");
+            return STC__FAILURE;
+        }
+
+        if ((s_server.hostentptr = gethostbyname(s_server.hostname)) == NULL) {
+            printf("--E--f_stc_createserver() error get local Internet address\n");
+            return STC__FAILURE;
+        }
+
+        retval = -1;
+
+        bzero((CHARS*)&s_server.sock_name, sizeof(s_server.sock_name));
+        s_server.sock_name.sin_family = AF_INET;
+        s_server.sock_name.sin_addr.s_addr = htonl(INADDR_ANY);
+        s_server.sock_name.sin_port = htons(*pl_port);
+
+        retval = bind(s_server.sock_rw, (struct sockaddr*)&s_server.sock_name, sizeof(s_server.sock_name));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'retval' is reassigned a value before the old one has been used.](https://app.codacy.com/manual/FairRootGroup/FairRoot/pullRequest?prid=5323596)